emcc.py

```diff
@@ -1249,6 +1249,11 @@ def default_setting(name, new_default):
       default_setting('IGNORE_MISSING_MAIN', 0)
       default_setting('DEFAULT_TO_CXX', 0)
 
+    # Default to TEXTDECODER=2 (always use TextDecoder to decode UTF-8 strings)
+    # in -Oz builds, since custom decoder for UTF-8 takes up space.
+    if shared.Settings.SHRINK_LEVEL >= 2 and not shared.Settings.USE_PTHREADS:
```

Comment:
why does pthreads matter? Is it that TextDecoder is not compatible with it? If so please link to the relevant issue (otherwise I worry this looks quite confusing...)

Reply:
Yes, that is the way Emscripten already behaves from before. Added comment to both places in code.